core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -566,12 +574,10 @@ object Fiber {
       final def await: UIO[Exit[E, A]]                     = IO.succeed(exit)
```

Comment:
We still have code doing `new Fiber[E, A]`, whereas, we want every fiber to be either `Runtime` or `Synthetic`. So we should change every instance of `new Fiber` in this file to be `new Synthetic`.

Then make sure `Fiber` is sealed in this file, so we won't have to worry about outside code doing the wrong thing.

Reply:
Yes. If we want to seal it I think we are going to need to move `FiberContext` into the same file.

Reply:
Used a trick to get around this.